src/dist/replication/meta_server/meta_bulk_load_service.cpp

```diff
@@ -572,8 +572,84 @@ void bulk_load_service::handle_app_ingestion(const bulk_load_response &response,
 void bulk_load_service::handle_bulk_load_finish(const bulk_load_response &response,
                                                 const rpc_address &primary_addr)
 {
-    // TODO(heyuchen): TBD
-    // called by `on_partition_bulk_load_reply` when app status is succeed, failed, canceled
+    const std::string &app_name = response.app_name;
+    const gpid &pid = response.pid;
+
+    if (!response.__isset.is_group_bulk_load_context_cleaned_up) {
```

Comment:
It is possible, I have met this condition during test, it is okay to keep it warning even if this condition is not happened.

Reply:
Under what conditions does `is_group_bulk_load_context_cleaned_up` may not set?

Reply:
In normal case, it won't happen, I meet it during kill test, I cannot remember the exact condition, but I think it's okay to keep this check here.

Reply:
When bulk load is canceled when some partitions have already finished, while this condition is not included this pull request.

Reply:
I have met this condition during test, I cannot remember the exact condition. I will continually test the code, if I confirm the condition won't happen I will update it, if not it is okay to have a check here.

Reply:
Whether `is_group_bulk_load_context_cleaned_up ` could be true or false depends on your code, if you're sure you have set it in all cases, I think you can use dassert instead.
Using assert instead of defensive check could be more clear and simple for logic and code. If a false assert happened, there must a bug and we have to fix it, not avoid it.

Reply:
I agree with you~